### PR TITLE
pkg/spiffs: implement statvfs and improve doc

### PR DESCRIPTION
**Pull request by vincent-d**
This implements the `statvfs` fs ops. The shell command `vfs df` can show file system usage.

Documentation is also improved.

### CODE REVIEW

**Comment by vincent-d** [comment]
If someone could take a look to this PR, it would be great!

**Comment by vincent-d** [comment]
I'd like to have that merge, could someone take a look?

**Comment by vincent-d** [comment]
It would be good to have this one in the release? could someone review? @aabadie @gebart ?

**Comment by vincent-d** [comment]
Ping!

**Comment by vincent-d** [comment]
Added a simple unittest. Another way to test is to have a SPIFFS filesystem mounted (you need a `mtd` device, so either native a spi flash; and the spiffs descriptor in your app or board config) and use the shell command `vfs df`.

**Comment by aabadie** [comment]
Looking at the code, the changes are fine but I don't have any hardware for testing this. Maybe @gebart, you could give it a try ?

**Comment by gebart** [comment]
I will test this during next week

**Comment by vincent-d** [comment]
@gebart did you find time to test this?

**Comment by gebart** [comment]
Sorry, I forgot about this PR. Will test tomorrow during the day. Ping me if I forget.

File: tests/unittests/tests-spiffs/tests-spiffs.c [hunk 1]
```diff
@@ -326,6 +326,37 @@ static void tests_spiffs_rename(void)
...
+
+    res = vfs_close(fd);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_statvfs("/test-spiffs/", &stat2);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    TEST_ASSERT_EQUAL_INT(1, stat2.f_bsize);
+    TEST_ASSERT_EQUAL_INT(1, stat2.f_frsize);
+    TEST_ASSERT(sizeof(buf) <= stat1.f_bfree - stat2.f_bfree);
```
Comment (gebart) [inline tests/unittests/tests-spiffs/tests-spiffs.c]: put some parentheses around the subtraction

File: tests/unittests/tests-spiffs/tests-spiffs.c [hunk 2]
```diff
@@ -326,6 +326,37 @@ static void tests_spiffs_rename(void)
...
+{
+    const char buf[] = "TESTSTRING";
+    struct statvfs stat1;
+    struct statvfs stat2;
+
+    int res = vfs_statvfs("/test-spiffs/", &stat1);
+    TEST_ASSERT_EQUAL_INT(0, res);
+    TEST_ASSERT_EQUAL_INT(1, stat1.f_bsize);
+    TEST_ASSERT_EQUAL_INT(1, stat1.f_frsize);
+    TEST_ASSERT(_dev->pages_per_sector * _dev->page_size * _dev->sector_count >=
```
Comment (gebart) [inline tests/unittests/tests-spiffs/tests-spiffs.c]: parens

File: tests/unittests/tests-spiffs/tests-spiffs.c [hunk 3]
```diff
@@ -326,6 +326,37 @@ static void tests_spiffs_rename(void)
...
+    res = vfs_close(fd);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_statvfs("/test-spiffs/", &stat2);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    TEST_ASSERT_EQUAL_INT(1, stat2.f_bsize);
+    TEST_ASSERT_EQUAL_INT(1, stat2.f_frsize);
+    TEST_ASSERT(sizeof(buf) <= stat1.f_bfree - stat2.f_bfree);
+    TEST_ASSERT(sizeof(buf) <= stat1.f_bavail - stat2.f_bavail);
```
Comment (gebart) [inline tests/unittests/tests-spiffs/tests-spiffs.c]: parens

**Comment by gebart** [comment]
Tests run fine on Mulle as well

**Comment by vincent-d** [comment]
Fixed parenthesis and amended directly

**Comment by gebart** [comment]
Murdock is green -> go

**Pull request merged**